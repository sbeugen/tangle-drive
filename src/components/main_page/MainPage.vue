<template>
  <div class="main-container">
    <nav-bar class="nav-bar"></nav-bar>
    <img class="iota-logo" src="@/assets/Iota_logo.png" alt="IOTA Logo">
    <p>Tangle-Drive can be used to upload files to the Tangle to share them securely.</p>
    <p>You simply select a file and receive a bundle hash which is needed to find and download your uploaded file.</p>
    <br>
    <p>As we are using 0 Value IOTA Transactions your files will be deleted during the next Snapshot.</p>
    <p>We are already working on a version of Tangle-Drive which allows you to store your files permanently.</p>
    <br>
    <button-group></button-group>
    <br>
    <br>
    <upload-section v-show="getUploadActive"></upload-section>
  </div>
</template>

<script>
import NavBar from '../NavBar'
import ButtonGroup from './ButtonGroup'
import UploadSection from './upload_section/UploadSection'

import { mapGetters } from 'vuex'

export default {
  components: {
    NavBar,
    ButtonGroup,
    UploadSection
  },
  computed: {
    ...mapGetters('upload', [
      'getUploadActive'
    ])
  }
}
  
</script>

<style scoped>
  .main-container {
    height: 100%;
    width: 100%;
  }
  .nav-bar {
    position: fixed;
    top: 0;
    right: 0;
    margin-right: 20px;
  }
  .iota-logo {
    filter: invert(100%);
    height: 100px;
    margin-top: 150px;
    margin-bottom: 50px;
  }
  button-group {
    margin-top: 50px;
    margin-bottom: 50px;
  }
</style>
