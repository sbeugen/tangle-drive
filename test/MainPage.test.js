import { shallow } from 'vue-test-utils'

import MainPage from '../src/components/main_page/MainPage'

describe('MainPage.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(MainPage)
  })

  it('has expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})