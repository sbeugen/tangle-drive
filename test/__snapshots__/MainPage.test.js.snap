// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage.test.js has expected html structure 1`] = `
<div
  class="main-container"
>
  <!---->
   
  <!---->
   
  <img
    alt="IOTA Logo"
    class="iota-logo"
    src="@/assets/tangle_drive_beta_logo.png"
  />
   
  <p
    class="made-by"
  >
    Made by 
    <a
      class="profile"
      href="https://github.com/sbeugen"
      target="_blank"
    >
      sbeugen
    </a>
  </p>
   
  <p>
    Tangle-Drive can be used to upload files to the 
    <a
      href="https://ipfs.io/"
      target="_blank"
    >
      IPFS
    </a>
     to share them securely.
  </p>
   
  <p>
    The content hash is stored in the 
    <a
      href="https://iota.org/"
      target="_blank"
    >
      IOTA Tangle
    </a>
    . You are able to access it by the bundle hash you receive after the upload.
  </p>
   
  <p>
    The 
    <a
      href="https://iota.org/"
      target="_blank"
    >
      Tangle
    </a>
     is an immutable storage. Therefore using this bundlehash, you can be 100% sure, that your file was not edited since the upload.
  </p>
   
  <br />
   
  <p>
    We are using 0 value 
    <a
      href="https://iota.org/"
      target="_blank"
    >
      IOTA
    </a>
     transactions. Thus your files will be deleted during the next Snapshot.
  </p>
   
  <p>
    Before this can happen, the 
    <a
      href="https://ipfs.io/"
      target="_blank"
    >
      IPFS
    </a>
     garbage collector will delete your files after aprox. 24 hours.
  </p>
   
  <p>
    We are already testing different solutions which will solve both problems.
  </p>
   
  <p>
    Also note that your files are not encrypted by Tangle-Drive. Anyone who has the bundle hash is able to download your files.
  </p>
   
  <br />
   
  <!---->
   
  <br />
   
  <br />
   
  <router-view />
</div>
`;
